lib/redix/start_options.ex

```diff
@@ -47,8 +46,19 @@ defmodule Redix.StartOptions do
     options
   end
 
-  defp fill_default_log_options(options) do
-    Keyword.update!(options, :log, &Keyword.merge(@default_log_options, &1))
+  defp warn_deprecated_log_option_and_add_defaults(opts) do
+    case Keyword.fetch(opts, :log) do
+      {:ok, log_opts} ->
+        IO.warn(
+          "The :log option has been deprecated in favour of using :telemetry. " <>
+            "See https://hexdocs.pm/redix/telemetry.html for more information."
```

Comment:
It is totally fine to deprecate the :log option but another idea is to always attach your own telemetry handler and check the log option in there.

This way your users get the logging out of the box and they still have a way to disable it and you keep the handler always running.

Reply:
You mean store the log option somewhere global (like app env) and access everytime I handle an event? Because right now the log option can be per connection.

Reply:
You can pass it as metadata too.

Reply:
@josevalim I've been thinking about this and there are pros and cons I think. The pros:

* users don't have to call `Redix.attach_default_telemtry_handler/0` in order to get logging out of the box

The cons:

* I want to deprecate `:log` since I don't like supporting the weird `event: :level` API that that option provides today.
* If users don't want logging, either I provide a way to detach the handler or I can't deprecate `:log` and let them use `log: false`.

So I'm inclined to go with what I have today. Wdyt?